atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraServer.java

```diff
@@ -18,30 +18,27 @@
 
 import com.palantir.logsafe.Preconditions;
 import java.net.InetSocketAddress;
-import java.util.List;
+import java.util.Set;
 import org.immutables.value.Value;
 
 @Value.Immutable
 public interface CassandraServer {
     @Value.Parameter
-    InetSocketAddress cassandraHostAddress();
+    String cassandraHostName();
 
     /**
-     * We do maintain a list of all IPs but do not create a client pool for each one of these. As of now this list is
-     * unused.
+     * {@code cassandraHostName()} with {@code reachableProxyIps()} form one reachable Cassandra server.
+     * While we maintain set of all IPs but do not create a client pool for each one of these.
      * */
-    @Value.Auxiliary
-    List<InetSocketAddress> reachableProxyIps();
+    Set<InetSocketAddress> reachableProxyIps();
 
     /**
      * The only proxy that will be used to reach the Cassandra host.
-     *
-     * We are making the assumption here that the list of IPs for a host will be consistent.
-     * In case this does not happen, we will black list this host as it will not be reachable using the proxy.
      * */
     @Value.Lazy
     default InetSocketAddress proxy() {
-        return reachableProxyIps().get(0);
+        // we know the set of proxies contains at least one element
+        return reachableProxyIps().stream().findFirst().get();
```

Comment:
```suggestion
        return reachableProxyIps().stream().findFirst().orElseThrow();
```

Reply:
Also, how about adding a `@Value.Check` to disallow setting `reachableProxyIps` to the empty set?

Reply:
we have that :)